bots/sg_rules_delete_by_scope.py

```diff
@@ -0,0 +1,124 @@
+    ip = re.split('/|\.', str(rule[SCOPE]))  # break ip to blocks
+    rule[SCOPE] = ip[0] + '.' + ip[1] + '.' + ip[2] + '.' + ip[3] + '/' + ip[4]
+    pass
+
+
+"""
+creates & removes the specified rules from a security group 
+"""
+
+
+def delete_sg(sg, sg_id, rule,direction, text_output):
+    # make sure that scope is in CIDR notation for example, 203.0.113.0/24
+    fix_malformed(rule)
+
+    if direction == 'inbound':
+        try:
+            sg.revoke_ingress(
+                CidrIp=rule[SCOPE],
+                FromPort=rule[PORT_FROM],
+                ToPort=rule[PORT_TO],
+                GroupId=sg_id,
+                IpProtocol=rule[PROTOCOL].lower()
+            )
+            text_output = text_output + ' rule : ' + rule[SCOPE] + ',' + str(rule[PORT_FROM]) + ',' + str(
+                rule[PORT_TO]) + ',' + \
+                          str(sg_id) + ',' + rule[PROTOCOL].lower() + ' deleted successfully ;'
+
+        except Exception as e:
+            text_output = text_output + f'Error while trying to delete security group. Error: {e}'
+
+    elif direction == 'outbound':
+        try:
+            sg.revoke_egress(
+                IpPermissions=[  # only IpPermissions supported with this func !
+                    {
+                        'FromPort': rule[PORT_FROM],
+                        'IpProtocol': rule[PROTOCOL].lower(),
+                        'IpRanges': [
+                            {
+                                'CidrIp': rule[SCOPE]
+                            },
+                        ],
+                        'ToPort': rule[PORT_TO]
+                    },
+                ]
+            )
+            text_output = text_output + ' rule : ' + rule[SCOPE] + ',' + str(rule[PORT_FROM]) + ',' + str(
+                rule[PORT_TO]) + ',' + str(sg_id) + ',' + rule[PROTOCOL].lower() + ' deleted successfully ;'
+
+        except Exception as e:
+            text_output = text_output + f'Error while trying to delete security group. Error: {e}'
+
+    else:
+        text_output = text_output + f'Error unknown direction ; \n'
+
+    return text_output
+
+
+def run_action(boto_session, rule, entity, params):
+    text_output = 'Run bot sg_rules_delete_by_scope.'
+    sg_id = entity['id']
+    # Param retrieving
+    try:
+        scope, direction, port, protocol = params  # get params
+    except Exception as e:
+        return e
+
+    ec2_resource = boto_session.resource('ec2')
+    sg = ec2_resource.SecurityGroup(sg_id)
+    port = str(port)
+    for rule in entity[f'{direction}Rules']:
+        if scope == rule[SCOPE]:
+            if port == '*' or int(port) == rule[PORT_FROM] == rule[PORT_TO]:
+                if protocol == rule[PROTOCOL] or protocol == '*':
+                    if rule[PROTOCOL] == 'ALL':
+                        rule[PROTOCOL] = ALL_TRAFFIC_PROTOCOL  # '-1'
+                    text_output = text_output + ' rule was found in security group with port in range ;'
```

Comment:
add some information about the rule like name or whatever

Reply:
done